twistlock/datadog_checks/twistlock/twistlock.py

```diff
@@ -82,8 +83,18 @@ def report_license_expiration(self):
             self.service_check(service_check_name, AgentCheck.CRITICAL, tags=self.config.tags)
             raise e
 
+        expiration_date_str = license.get("expiration_date")
+
+        # %f only matches microseconds. This will look for milliseconds and drop the last 3 digits.
+        match_milliseconds = re.match('(.*\.\d{6})(?:\d+)(Z)', expiration_date_str)
+        if match_milliseconds:
+            expiration_date_str = ""
+            for group in match_milliseconds.groups():
+                expiration_date_str += group
+
+        expiration_date = datetime.strptime(expiration_date_str, LICENSE_DATE_FORMAT)
```

Comment:
Hey, do you mean we need to support multiple formats ? or the initial format is wrong ?

If we do need to support multiple format, here are few ideas we can borrow: https://stackoverflow.com/questions/23581128/how-to-format-date-string-via-multiple-formats-in-python

IMHO we can avoid the custom parsing/matching `match_milliseconds = re.match('(.*\.\d{6})(?:\d+)(Z)', expiration_date_str)`

Reply:
supporting multiple formats might be a good idea.
Actually it may be a better idea to use datutil entirely for this. I'll rework this in another PR and close this one.